src/vorta/keyring/secretstorage.py

```diff
@@ -15,55 +14,55 @@ class VortaSecretStorageKeyring(VortaKeyring):
+            logger.debug("SecretStorage provider or DBus daemon is not available.")
+            raise e
+        else:
+            asyncio.set_event_loop(asyncio.new_event_loop())
+            self.collection = secretstorage.get_default_collection(self.connection)
 
     def set_password(self, service, repo_url, password):
         """
         Writes a password to the underlying store.
         """
-        try:
+        if self.is_unlocked:
             asyncio.set_event_loop(asyncio.new_event_loop())
-            collection = secretstorage.get_default_collection(self.connection)
             attributes = {
                 'application': 'Vorta',
                 'service': service,
                 'repo_url': repo_url,
                 'xdg:schema': 'org.freedesktop.Secret.Generic'}
-            collection.create_item(LABEL_TEMPLATE.format(repo_url=repo_url),
-                                   attributes,
-                                   password,
-                                   replace=True)
-        except secretstorage.exceptions.ItemNotFoundException:
-            logger.error("SecretStorage writing failed", exc_info=sys.exc_info())
+            self.collection.create_item(LABEL_TEMPLATE.format(repo_url=repo_url),
+                                        attributes,
+                                        password,
+                                        replace=True)
 
     def get_password(self, service, repo_url):
         """
         Retrieve a password from the underlying store. Return None if not found.
         """
         if self.is_unlocked:
             asyncio.set_event_loop(asyncio.new_event_loop())
-            collection = secretstorage.get_default_collection(self.connection)
             attributes = {'application': 'Vorta', 'service': service, 'repo_url': repo_url}
-            items = list(collection.search_items(attributes))
+            items = list(self.collection.search_items(attributes))
             logger.debug('Found %i passwords matching repo URL.', len(items))
             if len(items) > 0:
-                return items[0].get_secret().decode("utf-8")
+                item = items[0]
+                if item.is_locked() and item.unlock():
+                    return None
+                return item.get_secret().decode("utf-8")
         return None
 
     @property
     def is_unlocked(self):
-        try:
-            collection = secretstorage.get_default_collection(self.connection)
-            if collection.is_locked():  # Prompt for unlock
-                collection.unlock()
-            return not collection.is_locked()  # In case of denial
-        except secretstorage.exceptions.SecretServiceNotAvailableException:
-            logger.debug('SecretStorage is closed.')
+        # unlock() will return True if the unlock prompt is dismissed
+        if self.collection.is_locked() and self.collection.unlock():
             return False
+        return True
```

Comment:
```suggestion
        return not (self.collection.is_locked() and self.collection.unlock())
```
This is just a matter of preference, you can also keep the current code.

Reply:
You're right. It's obvious that it can be simplified to your solution.